src/CattyTests/Bricks/BrickUserDataTests.swift

```diff
@@ -111,4 +111,62 @@ final class BrickWithUserVariableTests: XCTestCase {
         XCTAssertTrue(addBrick.isVarOrListBeingUsed(userList))
         XCTAssertTrue(deleteBrick.isVarOrListBeingUsed(userList))
     }
+
+    func testUserVariableNoListUsedMultipleVariableBrick() {
+        let userVariable = UserVariable(name: "testName")
+        userVariable.value = "testValue"
+
+        let changeVariableBrick = ChangeVariableBrick()
+        let hideTextBrick = HideTextBrick()
+        let showTextBrick = ShowTextBrick()
+
+        changeVariableBrick.setVariable(userVariable, forLineNumber: 1, andParameterNumber: 1)
+        XCTAssertTrue(changeVariableBrick.isVarOrListBeingUsed(userVariable))
+        XCTAssertFalse(hideTextBrick.isVarOrListBeingUsed(userVariable))
+        XCTAssertFalse(showTextBrick.isVarOrListBeingUsed(userVariable))
+
+        hideTextBrick.setVariable(userVariable, forLineNumber: 1, andParameterNumber: 1)
+        XCTAssertTrue(changeVariableBrick.isVarOrListBeingUsed(userVariable))
+        XCTAssertTrue(hideTextBrick.isVarOrListBeingUsed(userVariable))
+        XCTAssertFalse(showTextBrick.isVarOrListBeingUsed(userVariable))
+
+        showTextBrick.setVariable(userVariable, forLineNumber: 1, andParameterNumber: 1)
+        XCTAssertTrue(changeVariableBrick.isVarOrListBeingUsed(userVariable))
+        XCTAssertTrue(hideTextBrick.isVarOrListBeingUsed(userVariable))
+        XCTAssertTrue(showTextBrick.isVarOrListBeingUsed(userVariable))
+    }
+
+    func testUserVariableNoListForBrickWithMultipleFormula() {
+        let userVariable = UserVariable(name: "testName", isList: false)
+
+        let brick = SetVariableBrick()
+        brick.setFormula(Formula(double: 50.50), forLineNumber: 1, andParameterNumber: 1)
+        brick.setFormula(Formula(string: "TestFormula"), forLineNumber: 2, andParameterNumber: 1)
+        brick.setFormula(Formula(integer: 100), forLineNumber: 2, andParameterNumber: 1)
+
+        brick.setVariable(userVariable, forLineNumber: 1, andParameterNumber: 1)
+
+        XCTAssertTrue(brick.isVarOrListBeingUsed(userVariable))
+    }
+
+    func testUserVariableNoListUseOnlyFormulaBrick() {
+        let userVariable = UserVariable(name: "testName", isList: false)
+
+        let brick = ArduinoSendDigitalValueBrick()
+        let formulaElemet = FormulaElement()
+        formulaElemet.value = userVariable.name
+        formulaElemet.type = ElementType.USER_VARIABLE
+
+        brick.setFormula(Formula(formulaElement: formulaElemet), forLineNumber: 1, andParameterNumber: 1)
```

Comment:
Please add an _assert false_ before this line to verify that before the state was different and correct the typo `formulaElemet`.